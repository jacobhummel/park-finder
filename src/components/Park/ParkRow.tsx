import { Park } from "api/parks";
import { Button, HStack, VStack, Spacer, Text } from "native-base";

interface Props extends Park {
  onPress(id: string): void;
}

const ParkRow = ({ id, name, address, onPress }: Props) => {
  return (
    <Button
      borderBottomWidth="1"
      _dark={{
        backgroundColor: "dark.400",
        borderColor: "gray.600",
      }}
      borderColor="coolGray.200"
      backgroundColor="white"
      justifyContent="flex-start"
      onPress={() => onPress(id)}
      py="3"
      px="4"
    >
      <HStack space={3} justifyContent="space-between">
        <VStack>
          <Text
            _dark={{
              color: "warmGray.50",
            }}
            color="coolGray.800"
            bold
          >
            {name}
          </Text>
          <Text
            color="coolGray.600"
            _dark={{
              color: "warmGray.200",
            }}
          >
            {address}
          </Text>
        </VStack>
        <Spacer />
      </HStack>
    </Button>
  );
};

export default ParkRow;
