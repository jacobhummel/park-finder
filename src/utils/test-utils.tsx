import { AppCtxProvider } from "../context";
import React from "react";
import { render as testingLibRender } from "@testing-library/react-native";
import { QueryClientProvider } from "react-query";
import { NavigationContainer } from "@react-navigation/native";
import { NativeBaseProvider } from "native-base";
import client from "api/client";
import customTheme from "constants/nativeBaseTheme";

export function render(element: React.ReactElement) {
  // https://docs.nativebase.io/next/testing
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  };

  return testingLibRender(
    <AppCtxProvider>
      <QueryClientProvider client={client}>
        <NavigationContainer theme={customTheme}>
          <NativeBaseProvider initialWindowMetrics={inset}>
            {element}
          </NativeBaseProvider>
        </NavigationContainer>
      </QueryClientProvider>
    </AppCtxProvider>
  );
}
