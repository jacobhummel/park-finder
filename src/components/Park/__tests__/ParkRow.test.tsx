import { fireEvent } from "@testing-library/react-native";
import { render } from "utils/test-utils";
import ParkRow, { ParkRowProps } from "../ParkRow";

test("should render", () => {
  // ARRANGE
  const testProps: ParkRowProps = {
    id: "1",
    name: "Parkers Lake Park",
    address: "15205 County Rd 6 & Niagara Lane, Plymouth, MN 55447",
    lat: 44.9964,
    lng: -93.4747,
    onPress: jest.fn(),
  };

  // ACT
  const { toJSON } = render(<ParkRow {...testProps} />);

  // ASSERT
  expect(toJSON()).toMatchSnapshot();
});

test("should pass id on click", () => {
  // ARRANGE
  const testProps: ParkRowProps = {
    id: "1",
    name: "Parkers Lake Park",
    address: "15205 County Rd 6 & Niagara Lane, Plymouth, MN 55447",
    lat: 44.9964,
    lng: -93.4747,
    onPress: jest.fn(),
  };

  // ACT
  const { getByRole } = render(<ParkRow {...testProps} />);

  const selectableRow = getByRole("button");
  fireEvent.press(selectableRow);

  // ASSERT
  expect(testProps.onPress).toBeCalledWith(testProps.id);
});
