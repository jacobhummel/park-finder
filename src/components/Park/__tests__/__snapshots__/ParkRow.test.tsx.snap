// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "borderColor": "#e5e7eb",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 12,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 12,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#1f2937",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Parkers Lake Park
            </Text>
            <View
              dataSet={Object {}}
              style={Object {}}
            />
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#4b5563",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              15205 County Rd 6 & Niagara Lane, Plymouth, MN 55447
            </Text>
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "width": 12,
              }
            }
          />
          <View
            dataSet={Object {}}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
